File: xmp/Locker.js
#!/usr/bin/env node
if ( require.main === module )
{
  var gepard = require ( "gepard" ) ;

  if ( gepard.getProperty ( "help" ) )
  {
    console.log (
      "Gepard example: Lock, lock a given resource.\n"
    + "Usage: node Lock [-Dname=<resource-name>], default <resource-name>=resid:main"
    ) ;
    process.exit() ;
  }

  new gepard.Admin().isRunning ( function admin_is_running ( state )
  {
    if ( ! state )
    {
      console.log ( "Not running on " + this.getHostPort() ) ;
      process.exit ( 1 ) ;
    }
    execute() ;
  });
  function execute()
  {
    var key = gepard.getProperty ( "name", "resid:main" ) ;
    var lock = new gepard.Lock ( key ) ;
    lock.acquire ( function ( err )
    {
      console.log ( "" + this.toString() ) ;
      if ( lock.isOwner() )
      {
        console.log ( "Sleep for 10 seconds" ) ;
        setInterval ( function sleep ()
        {
          lock.release() ;
          console.log ( "Lock released." ) ;
          process.exit() ;
        }, 10000 ) ;
      }
    } ) ;
  }
}

File: python/xmp/Locker.py
#!/usr/bin/env python

import os, sys
sys.path.insert ( 0, os.path.dirname(os.path.abspath(__file__) ) + "/../" )

from gepard import Client, Lock
import time

# ==========================================================================

client = Client.getInstance()

def on_error ( err, info ):
	print ( err )

client.onError ( on_error )

print ( "Try to lock resource='resid:main'" )
lock = Lock ( "resid:main" )
lock.acquire()

if lock.isOwner():
	print ( lock )
	print ( "Sleep for 10 seconds" )
	time.sleep(10)
	lock.release()
	print ( "Lock released." )
else:
	print ( lock )
